### ISSUE TITLE
Usb-c Hub 100w pass through shows "Battery is not charging"

### ISSUE THREAD

**Issue by jackreeves** [daliansky/XiaoMi-Pro#185]
I've bought 3 power/hd/usb hubs. Two rated at 65w and one rated at 100w for charging while using second monitor. All seem to slow charge and display message battery not charging. 

Using latest release and tested on two i7 16gb mx150's. Bios 960.

Not sure but seems to be software or Xiaomi related. Any insight @stevezhengshiqi ?

By the way. Thank you so much for your hard work. Trying to learn as much as I can to help you with this project. So much so I've swapped my 2010 and 2012 mbp to go all in!

**Comment by Menchen** [comment]
You shuold use a usb tester to check it is indeed charging... Xiaomi have implemented some kind of restrictions about PD, (limit 2A in some charger). Also booting to windows is the best way to test if it's software based..

**Comment by jackreeves** [comment]
Thanks menchen. In Windows it seems to move between charging and not charging in 3sec cycles. You can see the screen dim. Quite annoying actually. Glad I'm running OSX!
Don't have a usb tester. But hopefully these results help narrow down?

**Comment by jackreeves** [comment]
Main worry is that I'm ruining battery. It trickle charges enough to not take more than it receives so it is workable. Just this will be the main way I charge and doesn't seem like ideal behavior

**Comment by Menchen** [comment]
I think it's hardware issue and noting that we can fix.... try 65w one ..

**Comment by jackreeves** [comment]
Yeah tried two 65w ones. And this one is up to 100w... But couldn't find exact breakdown of voltage/amperage support. Anyone else using an external USBC hub with power they can recommend?

**Comment by keurfonluu** [comment]
I have the same issue with an Aukey USB-C hub using the original 65W charger (charging and not charging in Windows). I did not have any issue with the initial BIOS version (502), so it is probably related to this update.

**Comment by Giannidmg** [comment]
hi, I have been following this discussion for a while and I wanted to know if someone has found a solution to this problem
